src/components/Signup/Signup.js

```diff
@@ -105,12 +105,20 @@ const Signup = function ({
   const [alreadyExistentAddresses, setAlreadyExistentAddresses] = useState([]);
   const [blockedDomains, setBlockedDomains] = useState([]);
 
-  const utmSource = getSource(location);
-  const utmCampaign = getParameter(location, 'utm_campaign');
-  const utmMedium = getParameter(location, 'utm_medium');
-  const utmTerm = getParameter(location, 'utm_term');
+  let utmSource,
+    utmCampaign,
+    utmMedium,
+    utmTerm,
+    utmCookies = '';
```

Comment:
Why did you declare these variables out of _useEffect_ scope?

Reply:
I need this variables available outside the effect.

Reply:
What if you create a states for that variables?
const [utmCampaing, setUtmCampaign] = useState('');
...
an so on...

Reply:
or... 
`const [utmList, setUtmList] = useState({utmCampaign: '', utmSource: '' .....etc})`

```
useEffect(()=>{   
   setUtmList(
     {
        utmCampaign: getParameter(location, 'utm_campaing', 
        etc
     }
   )
}, []); 
```